src/bio/bio_xstream.c

```diff
@@ -337,12 +336,25 @@ bio_spdk_env_init(void)
 
 int
 bio_nvme_init(const char *nvme_conf, int shm_id, int mem_size,
-	      struct sys_db *db)
+	      int tgt_nr, struct sys_db *db)
 {
 	char		*env;
 	int		rc, fd;
 	uint64_t	size_mb = DAOS_DMA_CHUNK_MB;
 
+	D_ASSERT(tgt_nr > 0);
+	D_ASSERT(mem_size > 0);
+	/*
+	 * Hugepages are not enough to sustain average I/O workload
+	 * (~1GB per xstream).
+	 */
+	if ((mem_size / tgt_nr) < DAOS_DMA_MIN_UB_BUF_MB) {
+		D_ERROR("Per-xstream DMA buffer upperbound limit < 1GB!\n");
```

Comment:
```suggestion
		D_ERROR("Per-xstream DMA buffer upper bound limit < 1GB!\n");
```

Reply:
> > @ashleypittman I don't think the NLT test is actually using the nr_hugepages value in utils/nlt_server.yaml, still trying to investigate. Any ideas?
> 
> Server log file is here, and these seem to be the first two lines of interest:
> 
> https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5406/11/artifact/nlt_logs/dnt_server_h35n31mt.log/*view*/
> 
> 04/27-02:24:14.02 wolf-15 DAOS[33525/0/0] bio ERR src/bio/bio_xstream.c:352 bio_nvme_init() Per-xstream DMA buffer upperbound limit < 1GB!
> 04/27-02:24:14.02 wolf-15 DAOS[33525/0/0] bio DBUG src/bio/bio_xstream.c:354 bio_nvme_init() mem_size:256MB, DMA upper bound:64MB
> 
> Next time you've pushed could you merge master please as there are some reporting improvements in NLT that should show the error as a test failure. I also wonder which spelling of "upperbound" is correct?

Will fix spelling of upper bound - thanks for pointing that out. Also will rebase as well to get the new NLT updates.